src/SBSpergel.cpp

```diff
@@ -0,0 +1,524 @@
+        dy *= _r0;
+        dyx *= _r0;
+
+        It valit = val.linearView().begin();
+        for (int j=0;j<n;++j,x0+=dxy,y0+=dy) {
+            double x = x0;
+            double y = y0;
+            for (int i=0;i<m;++i,x+=dx,y+=dyx) {
+                double ksq = x*x + y*y;
+                *valit++ = _flux * _info->kValue(ksq);
+            }
+        }
+    }
+
+    SpergelInfo::SpergelInfo(double nu, const GSParamsPtr& gsparams) :
+        _nu(nu), _gsparams(gsparams),
+        _gamma_nup1(boost::math::tgamma(_nu+1.0)),
+        _gamma_nup2(_gamma_nup1 * (_nu+1)),
+        _gamma_nup3(_gamma_nup2 * (_nu+2)),
+        _gamma_mnum1(_nu < 0.0 ? boost::math::tgamma(-_nu-1.0) : 0.0), // only use this if nu<0
+        _xnorm0((_nu > 0.) ? _gamma_nup1 / (2. * _nu) * std::pow(2., _nu) : INFINITY),
+        _maxk(0.), _stepk(0.), _re(0.)
+    {
+        dbg<<"Start SpergelInfo constructor for nu = "<<_nu<<std::endl;
+
+        if (_nu < sbp::minimum_spergel_nu || _nu > sbp::maximum_spergel_nu)
+            throw SBError("Requested Spergel index out of range");
+    }
+
+    class SpergelIntegratedFlux
+    {
+    public:
+        SpergelIntegratedFlux(double nu, double gamma_nup2, double flux_frac=0.0)
+            : _nu(nu), _gamma_nup2(gamma_nup2),  _target(flux_frac) {}
+
+        double operator()(double u) const
+        {
+        // Return flux integrated up to radius `u` in units of r0, minus `flux_frac`
+        // (i.e., make a residual so this can be used to search for a target flux.
+        // This result is derived in Spergel (2010) eqn. 8 by going to Fourier space
+        // and integrating by parts.
+        // The key Bessel identities:
+        // int(r J0(k r), r=0..R) = R J1(k R) / k
+        // d[-J0(k R)]/dk = R J1(k R)
+        // The definition of the radial surface brightness profile and Fourier transform:
+        // Sigma_nu(r) = (r/2)^nu K_nu(r)/Gamma(nu+1)
+        //             = int(k J0(k r) / (1+k^2)^(1+nu), k=0..inf)
+        // and the main result:
+        // F(R) = int(2 pi r Sigma(r), r=0..R)
+        //      = int(r int(k J0(k r) / (1+k^2)^(1+nu), k=0..inf), r=0..R) // Do the r-integral
+        //      = int(R J1(k R)/(1+k^2)^(1+nu), k=0..inf)
+        // Now integrate by parts with
+        //      u = 1/(1+k^2)^(1+nu)                 dv = R J1(k R) dk
+        // =>  du = -2 k (1+nu)/(1+k^2)^(2+nu) dk     v = -J0(k R)
+        // => F(R) = u v | k=0,inf - int(v du, k=0..inf)
+        //         = (0 + 1) - 2 (1+nu) int(k J0(k R) / (1+k^2)^2+nu, k=0..inf)
+        //         = 1 - 2 (1+nu) (R/2)^(nu+1) K_{nu+1}(R) / Gamma(nu+2)
+            double fnup1 = std::pow(u / 2., _nu+1.)
+                * boost::math::cyl_bessel_k(_nu+1., u)
+                / _gamma_nup2;
+            double f = 1.0 - 2.0 * (1.+_nu)*fnup1;
+            return f - _target;
+        }
+    private:
+        double _nu;
+        double _gamma_nup2;
+        double _target;
+    };
+
+    class SpergelTaylorIntegratedFlux
```

Comment:
Right.  I was trying to use this to calculate the small radius enclosed flux before I discovered `solver.setXTolerance()`.

It's probably unlikely that we'd need to revive this, so I think I'll just get rid of it.  It will live on in the git history one way or another anyway.